bot/exts/help_channels/_cog.py

```diff
@@ -375,11 +377,17 @@ async def unclaim_channel(self, channel: discord.TextChannel, *, is_auto: bool =
             decorator = lock.lock_arg(f"{NAMESPACE}.unclaim", "claimant_id", wait=True)
             _unclaim_channel = decorator(_unclaim_channel)
 
-        return await _unclaim_channel(channel, claimant_id, is_auto)
+        return await _unclaim_channel(channel, claimant_id, closed_on)
+
+    async def _delete_message_time_caches(self, channel: discord.TextChannel) -> None:
+        """Delete message time caches."""
+        await _caches.claimant_last_message_times.delete(channel.id)
+        await _caches.non_claimant_last_message_times.delete(channel.id)
 
-    async def _unclaim_channel(self, channel: discord.TextChannel, claimant_id: int, is_auto: bool) -> None:
+    async def _unclaim_channel(self, channel: discord.TextChannel, claimant_id: int, closed_on: str) -> None:
         """Actual implementation of `unclaim_channel`. See that for full documentation."""
         await _caches.claimants.delete(channel.id)
+        await self._delete_message_time_caches(channel)
```

Comment:
Is the deletion still necessary? I hold the opinion that the fewer operations, the better. So if it can get away with leaving junk values in cache then it should just leave them. Or does that sound like a bad idea?